src/chat.rs

```diff
@@ -793,6 +793,14 @@ impl ChatId {
         context
             .sql
             .transaction(|transaction| {
+                let mut stmt = transaction.prepare("SELECT id FROM msgs WHERE chat_id=? AND type=?")?;
+                let rows = stmt.query_map((self, Viewtype::Webxdc), |row| row.get(0))?;
+                // Let's emit events optimistically (before the transaction completes) and not
+                // return webxdc `MsgId`s from the transaction because otherwise we need a vector of
+                // indefinite size in memory.
```

Comment:
while probably okay, i do not agree with the reasoning that a "a vector of indefinite size in memory" is an issue here.

i have a huge account, and if it is much in the largest chat (testing!) maybe 1000 webxdc. a vector of that size, and even if it is 1000x more is not an issue at all.

Reply:
I moved the code emitting events outside of the transaction. If the transaction becomes too huge, e.g. the chat contains too many messages, it doesn't scale well anyway and would block executing other transactions and the app, so let's not try to solve such problems in this PR